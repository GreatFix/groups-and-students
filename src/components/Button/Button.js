import React from 'react'
import classes from './Button.module.css'

const Button = (props) => {
  let color
  switch (props.color) {
    case 'danger':
      color = '#C9003F'
      break
    case 'info':
      color = '#00BABA'
      break
    case 'succes':
      color = '#40C81E'
      break
    default:
      color = '#0048BA'
  }
  return (
    <button
      disabled={props.disabled}
      className={classes.Button}
      onClick={props.onClick}
      style={{ backgroundColor: color }}
    >
      {props.children}
    </button>
  )
}

export default Button
