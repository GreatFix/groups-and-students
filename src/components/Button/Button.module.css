.Button {
  display: grid;
  justify-content: center;
  align-content: center;
  background-color: #70ce9b;
  box-sizing: border-box;
  border: 0;
  border-radius: 5px;
  margin: 2px;
  padding: 5px;
  outline: none;
  font-family: 'Lora', serif;
}
.Button:hover {
  box-shadow: 0px -0.25px 1px 1px#000;
}
