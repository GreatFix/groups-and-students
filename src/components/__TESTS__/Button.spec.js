import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Button from '../Button/Button'

const TEST_COLOR = 'success'
const TEST_TYPE = 'reset'
const TEST_CLASS = 'TEST_CLASS'
const TEST_TEXT = 'TEST_TEXT'
const TEST_VALUE = 'TEST_VALUE'

describe('Button', () => {
  test('render with children', () => {
    const { getByRole } = render(<Button>{TEST_TEXT}</Button>)

    expect(getByRole('button')).toBeInTheDocument()
    expect(getByRole('button')).toHaveTextContent(TEST_TEXT)
  })

  test('render with color', () => {
    const { getByRole } = render(<Button color={TEST_COLOR}>{TEST_TEXT}</Button>)
    expect(getByRole('button')).toHaveStyle(`background-color: #40C81E`)
  })

  test('render with type', () => {
    const { getByRole } = render(<Button type={TEST_TYPE}>{TEST_TEXT}</Button>)
    expect(getByRole('button')).toHaveAttribute('type', TEST_TYPE)
  })

  test('render with disabled', () => {
    const { getByRole } = render(<Button disabled>{TEST_TEXT}</Button>)
    expect(getByRole('button')).toBeDisabled()
  })

  test('render with className', () => {
    const { getByRole } = render(<Button className={TEST_CLASS}>{TEST_TEXT}</Button>)
    expect(getByRole('button')).toHaveClass(TEST_CLASS)
  })

  test('calls onClick', () => {
    const mockOnClick = jest.fn((e) => e.target.value)
    const { getByRole } = render(<Button onClick={mockOnClick}> {TEST_TEXT}</Button>)

    fireEvent.click(getByRole('button'), {
      target: {
        value: TEST_VALUE,
      },
    })

    expect(mockOnClick.mock.calls.length).toBe(1)
    expect(mockOnClick.mock.results[0].value).toBe(TEST_VALUE)
  })

  test('render properly', () => {
    const { getByRole } = render(
      <Button
        className={TEST_CLASS}
        color={TEST_COLOR}
        type={TEST_TYPE}
        onClick={() => {}}
        disabled
      >
        {TEST_TEXT}
      </Button>
    )
    expect(getByRole('button')).toMatchSnapshot()
  })
})
