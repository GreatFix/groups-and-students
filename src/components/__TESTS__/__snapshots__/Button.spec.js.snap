// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render properly 1`] = `
<button
  class="Button TEST_CLASS"
  disabled=""
  style="background-color: rgb(64, 200, 30);"
  type="reset"
>
  TEST_TEXT
</button>
`;
